tests/test_gabbits.py

```diff
@@ -0,0 +1,107 @@
+import json
+import os
+import random
+import socket
+import subprocess
+import sys
+import time
+
+import gabbi.driver
+import gabbi.fixture
+
+from gabbi.driver import test_pytest
+
+
+XSNIPPET_API_HOST = "localhost"
+XSNIPPET_API_PORT = 8000
+XSNIPPET_API_DATABASE_URL = os.getenv("ROCKET_DATABASE_URL")
+
+
+class XSnippetApi(gabbi.fixture.GabbiFixture):
+    """Start live server of XSnippet API."""
+
+    _launch_command = "cargo run"
+    _launch_timeout = 10.0
+    _shutdown_timeout = 5.0
+    _syntaxes = ["python", "rust", "clojure", "json", "lua"]
+
+    def __init__(self):
+        self.environ = {
+            "ROCKET_HOST": XSNIPPET_API_HOST,
+            "ROCKET_PORT": str(XSNIPPET_API_PORT),
+            "ROCKET_DATABASE_URL": XSNIPPET_API_DATABASE_URL,
+        }
+
+        if self._syntaxes:
+            # ROCKET_SYNTAXES expects a TOML array as input. Since TOML library
+            # for Python does not provide public means to serialize just a
+            # standalone array, we're relying on the fact that a JSON array of
+            # strings is fully compatible with a TOML array.
+            self.environ["ROCKET_SYNTAXES"] = json.dumps(self._syntaxes)
+
+        self.process = None
+
+    def start_fixture(self):
+        """Start the live server."""
+
+        environ = os.environ.copy()
+        environ.update(self.environ)
+
+        self.process = subprocess.Popen(self._launch_command, env=environ, shell=True)
```

Comment:
nit: passing `shell=True` adds a middleman between the test process and the xsnippet api running in the background, so the termination signal will have to go through shell process -> cargo process -> xsnippet api process. If termination works correctly now, then both shell and cargo forward the signals correctly, though, so maybe it's not a big deal.

Reply:
That's a nice point. I initially started with "shell=False" and changed it to `True` along the way trying to overcome various issue.